src/sql/platform/backup/common/backupService.ts

```diff
@@ -19,7 +19,7 @@ export const IBackupService = createDecorator<IBackupService>(SERVICE_ID);
 export interface IBackupService {
 	_serviceBrand: undefined;
 
-	getBackupConfigInfo(connectionUri: string): Thenable<azdata.BackupConfigInfo>;
+	getBackupConfigInfo(connectionUri: string): Promise<azdata.BackupConfigInfo | void>;
```

Comment:
In general we should switch to using Promise everywhere since mixing thenable and promise can complicate things from a code flow point of view, and theres really now point in abstracting this out. The only exception would be in our api layer were we don't want to enforce a PromiseLike implementation on extensions (Plus there are multiple different, contradicting, implementation of Promise) .